### PR TITLE
Support custom xcconfig file path

### PR DESCRIPTION
**Pull request by lpusok**
### Checklist
- [x] I've read and followed the [Contribution Guidelines](https://github.com/bitrise-steplib/.github/blob/main/CONTRIBUTING.md)
- [x] `step.yml` and `README.md` is updated with the changes (if needed)

### Version

Requires a *MINOR* [version update](https://semver.org/)

### Context

Feature request: https://github.com/bitrise-steplib/steps-xcode-archive/issues/273

Resolves: https://bitrise.atlassian.net/browse/STEP-1818

### Changes

Allow passing a file path to an existing `xcconfig` file, instead of providing the file contents directly.

### CODE REVIEW

File: vendor/github.com/bitrise-io/go-xcode/v2/xcconfig/xcconfig.go [hunk 1]
```diff
@@ -2,34 +2,57 @@ package xcconfig
...
 
 // NewWriter ...
-func NewWriter(pathProvider pathutil.PathProvider, fileManager fileutil.FileManager) Writer {
-	return &writer{pathProvider: pathProvider, fileManager: fileManager}
+func NewWriter(pathProvider pathutil.PathProvider, fileManager fileutil.FileManager, pathChecker pathutil.PathChecker) Writer {
+	return &writer{pathProvider: pathProvider, fileManager: fileManager, pathChecker: pathChecker}
 }
 
-func (w writer) Write(content string) (string, error) {
+// Write writes the contents of input into a xcconfig file if
```
Comment (shams-ahmed) [inline vendor/github.com/bitrise-io/go-xcode/v2/xcconfig/xcconfig.go:28]: Write writes

**Review by shams-ahmed — APPROVED** [review]

File: vendor/github.com/bitrise-io/go-xcode/v2/xcconfig/xcconfig.go (hunk 1 above)
Comment (lpusok) [reply vendor/github.com/bitrise-io/go-xcode/v2/xcconfig/xcconfig.go:28]: Can not fix it here, needs a go-xcode update.

File: main.go [hunk 2]
```diff
@@ -496,12 +511,14 @@ func (s XcodeArchiveStep) xcodeArchive(opts xcodeArchiveOpts) (xcodeArchiveOutpu
 		archiveCmd.SetCustomBuildAction("clean")
 	}
 
-	xcconfigWriter := xcconfig.NewWriter(s.pathProvider, s.fileManager)
-	xcconfigPath, err := xcconfigWriter.Write(opts.XcconfigContent)
-	if err != nil {
-		return out, fmt.Errorf("failed to write xcconfig file contents: %w", err)
+	if opts.XcconfigContent != "" {
+		xcconfigWriter := xcconfig.NewWriter(s.pathProvider, s.fileManager, s.pathChecker)
+		xcconfigPath, err := xcconfigWriter.Write(opts.XcconfigContent)
```
Comment (tothszabi) [inline main.go:516]: Just want to double check this part. This is the `Write` function from the config writer:
```
func (w writer) Write(input string) (string, error) {
	if w.isPath(input) {
		pathExists, err := w.pathChecker.IsPathExists(input)
		if err != nil {
			return "", fmt.Errorf(err.Error())
		}
		if !pathExists {
			return "", fmt.Errorf("provided xcconfig file path doesn't exist: %s", input)
		}
		return input, nil
	}

	dir, err := w.pathProvider.CreateTempDir("")
	if err != nil {
		return "", fmt.Errorf("unable to create temp dir for writing XCConfig: %v", err)
	}
	xcconfigPath := filepath.Join(dir, "temp.xcconfig")
	if err = w.fileManager.Write(xcconfigPath, input, 0644); err != nil {
		return "", fmt.Errorf("unable to write XCConfig content into file: %v", err)
	}
	return xcconfigPath, nil
}
```
If I read it correctly then if the user provided an xcconfig path as input then it returns relative and if not then it returns an absolute path. Can Xcode handle both types?

File: main.go [hunk 3]
```diff
@@ -226,6 +229,18 @@ func (s XcodeArchiveStep) ProcessInputs() (Config, error) {
...
+	var err error
+	config.XcodebuildAdditionalOptions, err = shellquote.Split(inputs.XcodebuildOptions)
+	if err != nil {
+		return Config{}, fmt.Errorf("provided XcodebuildOptions (%s) are not valid CLI parameters: %s", inputs.XcodebuildOptions, err)
+	}
+
+	config.XcconfigContent = strings.TrimSpace(config.XcconfigContent)
+	if sliceutil.IsStringInSlice("-xcconfig", config.XcodebuildAdditionalOptions) &&
+		config.XcconfigContent != "" {
+		return Config{}, fmt.Errorf("`-xcconfig` option found in XcodebuildOptions (`xcodebuild_options`), please clear Build settings (xcconfig) (`xcconfig_content`) input as only one can be set")
```
Comment (tothszabi) [inline main.go:241]: Nice to catch this and not leaving it for Xcode.

File: README.md [hunk 4]
```diff
@@ -29,7 +29,7 @@ To configure the Step:
 
 Under **xcodebuild configuration**:
 1. **Build configuration**: Specify Xcode Build Configuration. The Step uses the provided Build Configuration's Build Settings to understand your project's code signing configuration. If not provided, the Archive action's default Build Configuration will be used.
-2. **Build settings (xconfig)**: Build settings to override the project's build settings. The build settings must be separated by a newline character (`\n`).
+2. **Build settings (xcconfig)**: Build settings to override the project's build settings. Can be the contents, file path or empty.
```
Comment (tothszabi) [inline README.md:32]: Same relative path comment here too.

File: main.go (hunk 2 above)
Comment (lpusok) [reply main.go:516]: Checked both are supported by Xcode and both can be provided as input.
We could convert the xconfig path to absolute when parsing the Step inputs to be consistent tough.

File: step.yml [hunk 5]
```diff
@@ -121,20 +121,28 @@ inputs:
   opts:
     category: xcodebuild configuration
     title: Build settings (xcconfig)
-    summary: Build settings to override the project's build settings.
+    summary: Build settings to override the project's build settings. Can be the contents, file path or empty.
```
[the remark this answers is not in the document]
Comment (lpusok) [reply step.yml:123]: Technically can be an absolute path (but unlikely and not really reccomended).

File: README.md [hunk 6]
```diff
@@ -67,6 +67,29 @@ Add this step directly to your workflow in the [Bitrise Workflow Editor](https:/
 
 You can also run this step directly with [Bitrise CLI](https://github.com/bitrise-io/bitrise).
 
+### Examples
```
[the remark this answers is not in the document]
Comment (lpusok) [reply README.md:70]: Added a simple use case.

**Review by tothszabi — APPROVED** [review]

**Review by ofalvai — APPROVED** [review]

**Pull request merged**